packages/trie/test/trie/trie.spec.ts

```diff
@@ -1,104 +1,135 @@
 import { tmpdir } from 'os'
+import { mkdtempSync } from 'fs'
+import { join } from 'path'
 import { KECCAK256_RLP } from '@ethereumjs/util'
 import { Level } from 'level'
 import { MemoryLevel } from 'memory-level'
 import * as tape from 'tape'
 
-import { LevelDB, ROOT_DB_KEY, Trie } from '../../src'
-
-tape('Trie', function (t) {
-  t.test(
-    'creates an instance via the static constructor `create` function and defaults to `true` with a database',
-    async function (st) {
-      st.true(((await Trie.create({ db: new LevelDB(new Level(tmpdir())) })) as any)._persistRoot)
-    }
-  )
+import { CheckpointTrie, SecureTrie, Trie, LevelDB, ROOT_DB_KEY } from '../../src'
+
+for (const { constructor, title } of [
+  {
+    constructor: Trie,
+    title: 'Trie',
+  },
+  {
+    constructor: CheckpointTrie,
+    title: 'CheckpointTrie',
+  },
+  {
+    constructor: SecureTrie,
+    title: 'SecureTrie',
+  },
+]) {
+  let dbTmpDir: string
+  try {
+    dbTmpDir = mkdtempSync(join(tmpdir(), title))
+  } catch (error) {
+    dbTmpDir = tmpdir()
+  }
+
+  tape(`${title} (Persistence)`, function ({ test }) {
+    test('creates an instance via the static constructor `create` function and defaults to `true` with a database', async function (assert) {
```

Comment:
Sorry, this is what I was talking about.  For `tape` tests, our naming pattern across the monorepo is `tape( "some test", (t) => { (st) => { st.end() } t.end() }`.  I don't like calling subtests `assert` since that's too easily confused with the nodejs builtin.  Our usual pattern would be like below.  Can you apply this pattern (or something similar) across this test suite?
```suggestion
    t('creates an instance via the static constructor `create` function and defaults to `true` with a database', async (st) => {
```

Reply:
Great question.  If there is, it's woefully out of date as I certainly wasn't aware of anything when I started with the team.  I'll add a new issue for us to draft/update our contributor documentation as it's not something we've maintained (mostly since we've not had many outside contributors of late so yours are much much appreciated!).